src/pyqasm/modules/base.py

```diff
@@ -16,12 +16,17 @@
 Definition of the base Qasm module
 """
 
+from __future__ import annotations
+
+import functools
 from abc import ABC, abstractmethod
+from collections import Counter
 from copy import deepcopy
 from typing import Optional
 
 import openqasm3.ast as qasm3_ast
 from openqasm3.ast import BranchingStatement, Program, QuantumGate
+from tabulate import tabulate
```

Comment:
This should be imported lazily (i.e. outside top level, within the `compare` method) because it is not a core dependency, so will raise a `ModuleNotFoundError` if user does not have the `visualization` extra installed.